### DISCUSSION TITLE
ubuntu 20.04 no name resolution in client environment

### DISCUSSION THREAD

**Discussion by Reinemann67** [ltsp/ltsp#273] [Q&A]
Since I updated my server to ubuntu 20.04 there is no name resolution anymore in the client environment.
dig server  shows
;; connection timed out; no servers could be reached
cat /etc/resolv.conf  shows nothing

For other dhcp clients within the network the dns server is delivered fine
I already added a line into /etc/ltsp.conf 
DNS_SERVER="192.168.1.11 8.8.8.8 208.67.222.222"
but nothing happens

How should it work ?

**Comment by alkisg** [comment]
Hi, updated from which version? E.g. if you kept updating since 16.04, then you might have resolvconf installed, which shouldn't be used anymore.

What's the output of these commands on the client?

```shell
dpkg -l resolvconf
ls -l /etc/resolv.conf
cat /etc/resolv.conf
grep DNS_SERVER /etc/ltsp/ltsp.conf
systemd-resolve --status
```

**Comment by Reinemann67** [comment]
Hi alkisg,

I installed the new ltsp-github version on a ubuntu 19.10 end of 2019. 
Now I updated from that version to 20.04.
at my ltsp-client the following results were presented:

`dpkg -l resolvconf`
`Name           Version      Architektur  Beschreibung`
`+++-==============-============-============-=================================`
`ii  resolvconf     1.82         all          name server information handler`
`ls -l /etc/resolv.conf`
`lrwxrwxrwx 1 root root 29 Dez 27 15:54 /etc/resolv.conf -> ../run/resolvconf/resolv.conf`
`cat /etc/resolv.conf`
`=> nothing / nor a error, neither any content`
`grep DNS_SERVER /etc/ltsp/ltsp.conf`
`DNS_SERVER="192.168.1.11 8.8.8.8 208.67.222.222"`
`systemd-resolve --status`
`Failed to get global data: Unit dbus-org.freedesktop.resolve1.service not found.`

It seems I have to start the resolve / dns service anyhow, do I ?

Reine

**Comment by alkisg** [comment]
Reine, Ubuntu 19.10 was NOT using resolvconf. Maybe you installed some deprecated package that pulled it in, or maybe you followed some bad tutorial and you installed it.

The proper solution is to uninstall resolvconf, `apt purge --auto-remove resolvconf`, but before pressing "yes" there, do check which packages are getting autoremoved with it, because you may need some of them.

A second problem is that `systemd-resolve --status` doesn't work for you. This again might be because in the past you installed problematic packages, and removed some dependencies of systemd because they were conflicting. To resolve that, try to see if the problem happens on the server as well, and solve it there first.

Since you have multiple problems in your installation, maybe a clean install (while keeping /home) would be easier; depending on how bad it is.

**Comment by Reinemann67** [comment]
Hi,
I indeed installed a fresh 20.04 and the ltsp software.
Now it is systemd-resolve working and DNS for local an public addresses functional. Nevertheless I have to verify why. There are several config files and services I have to study.

Many thanks so far for your hints !
Reine
